resources/rebuild.sh

```diff
@@ -199,31 +201,121 @@ function build_al_kernel {
+    else 
+        KERNEL_VERSION=$1
+        if [[ "$KERNEL_VERSION" != @(5.10|5.10-no-acpi|6.1) ]]; then
+            die "Unsupported kernel version: '$KERNEL_VERSION'. Please use \`$0 help\` for help."
+        fi
+    fi
+
+    clone_amazon_linux_repo
 
-install_dependencies
+    # Apply kernel patches on top of AL configuration
+    apply_kernel_patches_for_ci
 
-BIN=overlay/usr/local/bin
-compile_and_install $BIN/init.c    $BIN/init
-compile_and_install $BIN/fillmem.c $BIN/fillmem
-compile_and_install $BIN/fast_page_fault_helper.c $BIN/fast_page_fault_helper
-compile_and_install $BIN/readmem.c $BIN/readmem
-if [ $ARCH == "aarch64" ]; then
-    compile_and_install $BIN/devmemread.c $BIN/devmemread
-fi
+    # Create the directory in which we will store the kernels
+    mkdir -p $OUTPUT_DIR
 
-build_rootfs ubuntu-22.04 jammy
-build_initramfs
+    if [[ "$KERNEL_VERSION" == @(all|5.10) ]]; then
+        build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10.config
+    fi
+    if [[ $ARCH == "x86_64" && "$KERNEL_VERSION" == @(all|5.10-no-acpi) ]]; then
+        build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10-no-acpi.config
+    fi
+    if [[ "$KERNEL_VERSION" == @(all|6.1) ]]; then
+        build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-6.1.config 5.10
+    fi
+}
 
-clone_amazon_linux_repo
+function build_non_ci_kernel {
+    local LINUX_REPO_URL=$1
+    local LINUX_KCONFIG_PATH=$2
 
-# Apply kernel patches on top of AL configuration
-apply_kernel_patches_for_ci
+    clone_linux_repo $LINUX_REPO_URL
+    build_al_kernel $LINUX_KCONFIG_PATH
+}
 
-build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10.config
-if [ $ARCH == "x86_64" ]; then
-    build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10-no-acpi.config
-fi
-build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-6.1.config
+function print_help {
+    echo ""
+    echo "Firecracker CI artifacts build script"
+    echo "Usage: $(basename $0) [<command>] [<command args>]"
+    echo ""
+    echo "Available commands:"
+    echo ""
+    echo "    all (default)"
+    echo "        Build CI rootfs and default guest kernels using configurations from"
+    echo "        resources/guest_configs."
+    echo "        This will patch the guest configurations with all the patches under"
+    echo "        resources/guest_configs/patches."
+    echo "        This is the default command, if no command is chosen."
+    echo ""
+    echo "    rootfs"
+    echo "        Builds only the CI rootfs."
+    echo ""
+    echo "    kernels [version]"
+    echo "        Builds our the currently supported CI kernels."
+    echo ""
+    echo "        version: Optionally choose a kernel version to build. Supported"
+    echo "                 versions are: 5.10, 5.10-no-acpi or 6.1."
+    echo ""
+    echo "    help"
+    echo "        Displays the help message and exits."
+    
+}
```

Comment:
Maybe just 

```
cat <<EOF
...
EOF
```

Reply:
Just followed the `devtool` style. I don't have preference, so if you prefer it like this, I'll do it :)

Reply:
did it anyway.